data/test/scenarios/test_movetype.cfg

```diff
@@ -0,0 +1,344 @@
+						[/destination]
+						allow_multiple_turns=no
+						variable=path
+					[/find_path]
+					# The path's steps include the starting hex, which is why there's
+					# a +1 here and a +1 in the [for] loop.
+					{ASSERT {VARIABLE_CONDITIONAL path.hexes not_equals 0}}
+					{ASSERT {VARIABLE_CONDITIONAL path.step.length equals 8}}
+					{VARIABLE expected_total 0}
+					[for]
+						array=expected_movement_cost
+						[do]
+							{VARIABLE i_plus_one "$($i + 1)"}
+							[set_variable]
+								name=expected_total
+								add=$expected_movement_cost[$i].$traveler.id
+							[/set_variable]
+							{ASSERT {VARIABLE_CONDITIONAL path.step[$i_plus_one].movement_cost equals $expected_total}}
+						[/do]
+					[/for]
+				[/do]
+			[/foreach]
+
+			{SUCCEED}
+		[/event]
+	[/test]
+#enddef
+
+# This test works by setting Alice's vision points, resetting the fog and then seeing how many hexes are visible.
+# The parameter ALICE_VISION_COSTS contains the values of each hex in the path from Alice to the destination,
+# so the same order as in the movement tests. Vision will also see the path towards Bob, for which this assumes that
+# the terrain adjacent to Bob's hex is the same as the terrain adjacent to Alice's.
+#
+# The vision cost of the location_id=destination hex will be ignored, as that hex becomes visible based on the cost
+# of the adjacent hex, however it's included in the macro argument for easier cut&paste from the MOVETYPE_MOVE_SCEN
+# tests.
+#define MOVETYPE_VISION_SCEN ID ALICE_TYPE ALICE_VISION_COSTS ALICE_MODIFICATION
+	[test]
+		name = "Unit Test {ID}"
+		map_data = "{test/maps/test_movetype.map}"
+		turns = 1
+		id = {ID}
+		random_start_time = no
+		is_unit_test = yes
+
+		{DAWN}
+
+		[side]
+			side=1
+			controller=human
+			name = "Alice"
+			type = {ALICE_TYPE}
+			id=alice
+			fog=yes
+			shroud=no
+			share_view=no
+		[/side]
+		[side]
+			side=2
+			controller=human
+			name = "Bob"
+			type = Orcish Grunt
+			id=bob
+			fog=yes
+			shroud=no
+			share_view=no
+		[/side]
+
+		[event]
+			name = prestart
+			[set_variables]
+				name=expected_vision_costs
+				mode=replace
+				[split]
+					list={ALICE_VISION_COSTS}
+					key=alice
+					separator=","
+				[/split]
+			[/set_variables]
+			{ASSERT {VARIABLE_CONDITIONAL expected_vision_costs.length equals 7}}
+
+			# This could be counted with a location filter, but other assumptions about the
+			# map already need to be hardcoded in the test.
+			{VARIABLE walkable_hexes_on_map 10}
+		[/event]
+
+		[event]
+			name = side 1 turn 1
+
+			{ALICE_MODIFICATION}
+
+			{VARIABLE vision_points 0}
+			{VARIABLE expected_visible_count 0}
+			[while]
+				{VARIABLE_CONDITIONAL expected_visible_count less_than $walkable_hexes_on_map}
+				{VARIABLE_CONDITIONAL ended boolean_equals no} # this is set by ASSERT
+				[do]
+					[modify_unit]
+						[filter]
+							id=alice
+						[/filter]
+						vision=$vision_points
+					[/modify_unit]
+					[reset_fog]
+						side=1
```

Comment:
This fails schema validation on 1.15, which is a build breaker. Will fix before this is ready for merging.